setup.py

```diff
@@ -10,6 +10,26 @@
     open('CHANGES.rst').read() + '\n'
 )
 
+TEST_REQUIRES = [
```

Comment:
I prefer to have the requirements inline and keep things as they are. This change makes the code less readable IMHO. This PR is already way too large to properly review, we should try to keep things like that out.

Reply:
From what I can see, it was changed like this so that the Python 2-only test requirements can be added in conditionally before passing to `setup()`. Archetypes does not have Python 3 support.

Reply:
ic. Though, right now they are not added conditionally right? This is a minor issue, so no long discussion needed either way...

Reply:
They are -- see below line 30-31

Reply:
Uh. Missed that. Maybe we should add some comments to make this more obvious. :)